src/test/app/Loan_test.cpp

```diff
@@ -0,0 +1,2710 @@
+  Copyright (c) 2025 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+
+#include <xrpld/app/misc/LendingHelpers.h>
+#include <xrpld/app/misc/LoadFeeTrack.h>
+#include <xrpld/app/tx/detail/LoanSet.h>
+
+#include <xrpl/beast/unit_test/suite.h>
+
+namespace ripple {
+namespace test {
+
+class Loan_test : public beast::unit_test::suite
+{
+    // Ensure that all the features needed for Lending Protocol are included,
+    // even if they are set to unsupported.
+    FeatureBitset const all{
+        jtx::testable_amendments() | featureMPTokensV1 |
+        featureSingleAssetVault | featureLendingProtocol};
+
+    static constexpr auto const coverDepositParameter = 1000;
+    static constexpr auto const coverRateMinParameter =
+        percentageToTenthBips(10);
+    static constexpr auto const coverRateLiquidationParameter =
+        percentageToTenthBips(25);
+    static constexpr auto const maxCoveredLoanValue = 1000 * 100 / 10;
+    static constexpr auto const vaultDeposit = 1'000'000;
+    static constexpr auto const debtMaximumParameter = 25'000;
+    std::string const iouCurrency{"IOU"};
+
+    void
+    testDisabled()
+    {
+        testcase("Disabled");
+        // Lending Protocol depends on Single Asset Vault (SAV). Test
+        // combinations of the two amendments.
+        // Single Asset Vault depends on MPTokensV1, but don't test every combo
+        // of that.
+        using namespace jtx;
+        auto failAll = [this](FeatureBitset features, bool goodVault = false) {
+            Env env(*this, features);
+
+            Account const alice{"alice"};
+            Account const bob{"bob"};
+            env.fund(XRP(10000), alice, bob);
+
+            auto const keylet = keylet::loanbroker(alice, env.seq(alice));
+
+            using namespace std::chrono_literals;
+            using namespace loan;
+
+            // counter party signature is optional on LoanSet. Confirm that by
+            // sending transaction without one.
+            auto setTx =
+                env.jt(set(alice, keylet.key, Number(10000)), ter(temDISABLED));
+            env(setTx);
```

Comment:
Why is `setTx` needed? Why not to call this way?

     env(set(alice, keylet.key, Number(10000)), ter(temDISABLED));